### PR TITLE
Moderators can specify the reason for moderation

### PR DESCRIPTION
**Pull request by gabehamilton**
When moderating content or an account, moderators are now presented with a dialog to choose the reason. This same dialog will be also used by users when reporting (future PR).

### CODE REVIEW

File: src/DIG/Dialog.jsx [hunk 1]
```diff
@@ -202,7 +202,7 @@ return (
         )}
       </ActionWrapper>
       {(type === "dialog" || enableCloseButton) && (
-        <CloseButtonWrapper>
+        <CloseButtonWrapper onClick={onCancel}>
```
Comment (shelegdmitriy) [inline src/DIG/Dialog.jsx:205]:
```suggestion
        <CloseButtonWrapper>
```

No need to pass another prop here. Just use [onOpenChange](https://www.radix-ui.com/primitives/docs/components/alert-dialog#root) in `ReasonDialog` component

File: src/Moderation/NeedsModeration.jsx [hunk 2]
```diff
@@ -289,19 +292,16 @@ const renderItem = (item) => {
...
                 tooltip: blockAccountHelperText + (isModerator ? "" : "\n" + disabledMessage),
                 variant: "destructive",
                 fill: "outline",
-                data: {
-                  index: {
-                    moderate: moderationDataFormat(accountId),
-                  },
-                },
+                type: pathToType("account"),
+                buildData: moderationDataFormat.bind(null, accountId, null, null),
```
Comment (shelegdmitriy) [inline src/Moderation/NeedsModeration.jsx:304]:
```suggestion
                buildData: moderationDataFormat(null, accountId, null, null),
```

I believe there is no need to use `.bind`

File: src/Moderation/ReasonDialog.jsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
+const type = props.type;
+const open = props.open;
+const closeModal = props.closeModal;
+const submitClick = props.submitClick;
```
Comment (shelegdmitriy) [inline src/Moderation/ReasonDialog.jsx:4]:
```suggestion
let {
  type,
  open,
  closeModal,
  submitClick
} = props;
```

File: src/Moderation/ReasonDialog.jsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+          <h4>Report {type}</h4>
+        </div>
+        <div className="d-flex flex-row gap-2">
+          <span>Tell us more about why this {type} is problematic.</span>
+        </div>
+      </div>
+      <div className="d-flex flex-column gap-3">
+        {Object.keys(reasons).map((reason) => {
+          return (
+            <div className="d-flex flex-row gap-2">
```
Comment (shelegdmitriy) [inline src/Moderation/ReasonDialog.jsx:69]: Please provide an unique `key` for the element:

```suggestion
            <div className="d-flex flex-row gap-2" key={reason.id // or what is the unique value here}>
```

File: src/Moderation/ReasonDialog.jsx [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+            </div>
+          );
+        })}
+      </div>
+      <div className="d-flex flex-row gap-2">
+        <InfoText>
+          <i className="ph-bold ph-info" /> Unsure? Review our community{" "}
+          <a href="https://near.org/terms" target="_blank">
+            content policies.
+          </a>
```
Comment (shelegdmitriy) [inline src/Moderation/ReasonDialog.jsx:94]:
```suggestion
          <Link href="https://near.org/terms" target="_blank">
            content policies.
          </Link>
```

File: src/Moderation/ReasonDialog.jsx [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+  font-size: 1em;
+`;
+
+return (
+  <Widget
+    src="${REPL_ACCOUNT}/widget/DIG.Dialog"
+    props={{
+      type: "dialog",
+      description: buildContent(type),
+      onCancel: closeModal,
```
Comment (shelegdmitriy) [inline src/Moderation/ReasonDialog.jsx:144]:
```suggestion
      onOpenChange: closeModal,
```

**Review by shelegdmitriy — CHANGES_REQUESTED** [review]

File: src/Moderation/NeedsModeration.jsx (hunk 2 above)
Comment (gabehamilton) [reply src/Moderation/NeedsModeration.jsx:304]: This function gets passed in to the dialog and called there with the reason.

Comment (shelegdmitriy) [reply src/Moderation/NeedsModeration.jsx:304]: Your method is written as an arrow function that's why no need to use `bind` there. Try that out 🙂

File: src/Moderation/ReasonDialog.jsx (hunk 4 above)
Comment (gabehamilton) [reply src/Moderation/ReasonDialog.jsx:66]: React isn't complaining about this for me since it's not a React component. Why do we need a key?

**Review by calebjacob — APPROVED** [review]
Looks good @gabehamilton - let's just make sure we address the `key` comment above before merging.

**Pull request merged**